File: solutions/serial_thread.c
#include <stdio.h>
#include <time.h>

#define NUM_HELLOS 200000

int main(){
    float startTime = (float) clock()/CLOCKS_PER_SEC;
    for(int i = 0; i <= NUM_HELLOS; i++){
        printf("Hello World x %d \r", i);
        fflush(stdout);
    }
    float endTime = (float) clock()/CLOCKS_PER_SEC;
    printf("Total Execution Time: %f", endTime-startTime);
}

File: solutions/hello-world/serial_thread.py
import time

NUM_HELLOS = 2000000

if __name__ == '__main__':
    t1 = time.time()
    for i in range(NUM_HELLOS):
        print('Hello World x', i, end='\r')
    t2 = time.time()

    print('Total Execution Time: ', t2-t1)